src/lib.rs

```diff
@@ -712,38 +714,40 @@ fn postprocess_chunks(
             !invalid
         });
     }
+}
 
-    // Find fdAT chunks and attempt to recompress them
-    // Note if there are multiple fdATs per frame then decompression will fail and nothing will change
-    let mut fdat: Vec<_> = png
-        .aux_chunks
-        .iter_mut()
-        .filter(|c| &c.name == b"fdAT")
-        .collect();
-    if opts.idat_recoding && !fdat.is_empty() {
-        let buffer_size = orig_ihdr.raw_data_size();
-        fdat.par_iter_mut()
-            .with_max_len(1)
-            .enumerate()
-            .for_each(|(i, c)| {
-                if deadline.passed() || c.data.len() <= 4 {
-                    return;
-                }
-                if let Ok(mut data) = deflate::inflate(&c.data[4..], buffer_size).and_then(|data| {
-                    let max_size = AtomicMin::new(Some(c.data.len() - 5));
-                    opts.deflate.deflate(&data, &max_size)
-                }) {
-                    debug!(
-                        "Recompressed fdAT #{:<2}: {} ({} bytes decrease)",
-                        i,
-                        c.data.len(),
-                        c.data.len() - 4 - data.len()
-                    );
-                    c.data.truncate(4);
-                    c.data.append(&mut data);
-                }
-            });
+/// Recompress the additional frames of an APNG
+fn recompress_frames(png: &mut PngData, opts: &Options, deadline: Arc<Deadline>) {
+    if !opts.idat_recoding || png.frames.is_empty() {
+        return;
     }
+    // Use the same filter chosen for the main image
+    let filter = png.filter.unwrap_or(RowFilter::None);
+    png.frames
+        .par_iter_mut()
+        .with_max_len(1)
+        .enumerate()
+        .for_each(|(i, frame)| {
+            if deadline.passed() {
+                return;
+            }
+            let mut ihdr = png.raw.ihdr.clone();
+            ihdr.width = frame.width;
+            ihdr.height = frame.height;
+            if let Ok(data) = PngImage::new(ihdr, &frame.data).and_then(|image| {
```

Comment:
Should we error here if decompression fails?

Reply:
I've now made this throw any error from `PngImage::new`, which I think this is the safest thing to do for now.

Reply:
Not anymore, it's just the way it was working before. It would try to recompress the fdATs but since we weren't consolidating them it would often fail.